src/main/java/net/openhft/chronicle/bytes/internal/Decimalizer.java

```diff
@@ -0,0 +1,217 @@
+     *
+     * @param value           the float value to be converted.
+     * @param decimalAppender the DecimalAppender used to store the converted decimal value.
+     * @return true if the conversion and appending was successful, false otherwise.
+     */
+    boolean toDecimal(float value, DecimalAppender decimalAppender);
+
+    /**
+     * The default implementation of Decimalizer which attempts the conversion using the light-weight strategy first,
+     * falling back to the BigDecimal-based strategy if necessary.
+     */
+    class Instance implements Decimalizer {
+        @Override
+        public boolean toDecimal(double value, DecimalAppender decimalAppender) {
+            return LITE.toDecimal(value, decimalAppender)
+                    || USES_BIG_DECIMAL.toDecimal(value, decimalAppender);
+        }
+
+
+        @Override
+        public boolean toDecimal(float value, DecimalAppender decimalAppender) {
+            return LITE.toDecimal(value, decimalAppender)
+                    || USES_BIG_DECIMAL.toDecimal(value, decimalAppender);
+        }
+    }
+
+    /**
+     * Light-weight implementation of Decimalizer, which uses a simple rounding approach for conversion.
+     */
+    class Lite implements Decimalizer {
+
+        public static final int LARGEST_EXPONENT_IN_LONG = 18;
+
+        /**
+         * Converts a double value to its decimal representation using a simple rounding approach and appends it.
+         *
+         * @param value           the double value to be converted.
+         * @param decimalAppender the DecimalAppender used to store the converted decimal value.
+         * @return true if the conversion and appending was successful, false otherwise.
+         */
+        public boolean toDecimal(double value, DecimalAppender decimalAppender) {
+            boolean isNegative = Double.doubleToLongBits(value) < 0;
+            double absValue = Math.abs(value);
+
+            long factor = 1;
+            for (int exponent = 0; exponent <= LARGEST_EXPONENT_IN_LONG; exponent++) {
+                long mantissa = Math.round(absValue * factor);
+                if ((double) mantissa / factor == absValue) {
+                    decimalAppender.append(isNegative, mantissa, exponent);
+                    return true;
+                }
+                factor *= 10;
+            }
+            return false;
+        }
+
+        /**
+         * Converts a float value to its decimal representation using a simple rounding approach and appends it.
+         *
+         * @param value           the float value to be converted.
+         * @param decimalAppender the DecimalAppender used to store the converted decimal value.
+         * @return true if the conversion and appending was successful, false otherwise.
+         */
+        public boolean toDecimal(float value, DecimalAppender decimalAppender) {
+            boolean sign = Double.doubleToLongBits(value) < 0;
```

Comment:
Use `Float.floatToIntBits(...)` to avoid the cast perhaps?

Reply:
Indeed.